indico/modules/events/registration/fields/choices.py

```diff
@@ -143,7 +143,10 @@ def _check_number_of_places(new_data):
                     places_used_dict.setdefault(k, 0)
                     if old_data and old_data.data:
                         places_used_dict[k] -= old_data.data.get(k, 0)
-                    places_used_dict[k] += new_data[k]
+                    try:
+                        places_used_dict[k] += new_data[k]
+                    except TypeError:
+                        raise ValidationError(_('Invalid data'))
```

Comment:
```suggestion
                        raise ValidationError('Invalid data')
```

No need for translations here, since this is not really a user-facing error